lib/cext/preprocess.rb

```diff
@@ -6,935 +6,120 @@
+
+      simple = []
+      arrays = []
+
+      line = $1
+
+      $2.split(',').each do |local|
+        local.strip!
+        if local.end_with?(']')
+          raise unless local =~ /(\w+)\s*\[\s*(\d+)\s*\]/
+          arrays << [$1, $2.to_i]
+        else
+          simple << local
+        end
+      end
 
-    arrays.each do |name, size|
-      line += " VALUE *#{name} = truffle_managed_malloc(#{size} * sizeof(VALUE));"
-    end
-    line
-  elsif line =~ ALLOCA_LOCALS
-    # Translate
-    #   VALUE *argv = alloca(sizeof(VALUE) * argc);
-    # into
-    #   VALUE *argv = (VALUE *)truffle_managed_malloc(sizeof(VALUE) * argc);
+      unless simple.empty?
+        line += "VALUE #{simple.join(', ')};"
+      end
 
-    line = "#{$1}VALUE *#{$2} = truffle_managed_malloc(#{$'}"
-  else
-    line
+      arrays.each do |name, size|
+        line += " VALUE *#{name} = truffle_managed_malloc(#{size} * sizeof(VALUE));"
+      end
+      line
+    elsif line =~ ALLOCA_LOCALS
+      # Translate
+      #   VALUE *argv = alloca(sizeof(VALUE) * argc);
+      # into
+      #   VALUE *argv = (VALUE *)truffle_managed_malloc(sizeof(VALUE) * argc);
+
+      line = "#{$1}VALUE *#{$2} = truffle_managed_malloc(#{$'}"
+    else
+      line
+    end
   end
-end
 
-def patch(file, contents, directory)
-  if patched_file = PATCHED_FILES[File.basename(file)]
-    matched = if patched_file[:ext_dir]
-                directory.end_with?(File.join(patched_file[:gem], 'ext', patched_file[:ext_dir]))
-              else
-                regexp = /^#{Regexp.escape(patched_file[:gem])}\b/
-                directory.split('/').last(2).any? { |part| part =~ regexp } || file.split('/').last(2).any? { |part| part =~ regexp }
-              end
-    if matched
-      patched_file[:patches].each do |patch|
-        contents = contents.gsub(patch[:match], patch[:replacement].rstrip)
+  def self.patch(file, contents, directory)
+    if patched_file = PATCHED_FILES[File.basename(file)]
+      matched = if patched_file[:ext_dir]
+                  directory.end_with?(File.join(patched_file[:gem], 'ext', patched_file[:ext_dir]))
+                else
+                  regexp = /^#{Regexp.escape(patched_file[:gem])}\b/
+                  directory.split('/').last(2).any? { |part| part =~ regexp } || file.split('/').last(2).any? { |part| part =~ regexp }
+                end
+      if matched
+        patched_file[:patches].each do |patch|
+          contents = contents.gsub(patch[:match], patch[:replacement].rstrip)
+        end
       end
     end
+    contents
   end
-  contents
-end
 
-if __FILE__ == $0
-  puts "#line 1 \"#{ARGF.filename}\""
+  if __FILE__ == $0
+    puts "#line 1 \"#{ARGF.filename}\""
 
-  contents = patch(ARGF.filename, File.read(ARGF.filename), Dir.pwd)
-  contents.each_line do |line|
-    puts preprocess(line)
+    contents = patch(ARGF.filename, File.read(ARGF.filename), Dir.pwd)
+    contents.each_line do |line|
+      puts preprocess(line)
+    end
   end
```

Comment:
This "if main" should be moved outside the class, could you do it in another PR since this one is merged?